grass7/raster/r.confusionmatrix/r.confusionmatrix.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env python
+
+############################################################################
+#
+# MODULE:	r.confusionmatrix
+#
+# AUTHOR(S):    Anika Bettge <bettge at mundialis.de>
+#
+# PURPOSE:      Calculates a confusion matrix and accuracies for a given classification using r.kappa.
+#
+# COPYRIGHT:	(C) 2020 by mundialis and the GRASS Development Team
+#
+#		This program is free software under the GNU General Public
+#		License (>=v2). Read the file COPYING that comes with GRASS
+#		for details.
+#
+#############################################################################
+
+#%Module
+#% description: Calculates a confusion matrix and accuracies for a given classification using r.kappa.
+#% keyword: raster
+#% keyword: statistics
+#% keyword: accuracy
+#% keyword: classification
+#%end
+
+#%option G_OPT_R_INPUT
+#% key: classification
+#% required: no
+#% description: Name of raster map containing classification result
+#%end
+
+#%option G_OPT_R_INPUT
+#% key: raster_reference
+#% required: no
+#% description: Name of raster map containing reference classes
+#%end
+
+#%option G_OPT_V_INPUT
+#% key: vector_reference
+#% required: no
+#% description: Name of vector map containing reference classes
+#%end
+
+#%option G_OPT_DB_COLUMN
+#% key: column
+#% required: no
+#% description: Name of column in the vector map containing reference classes
+#%end
+
+#%option
+#% key: label_column
+#% type: string
+#% required: no
+#% multiple: no
+#% key_desc: name
+#% description: Name of column used as raster category labels
+#%end
+
+#%option G_OPT_F_OUTPUT
+#% key: csvfile
+#% required: no
+#% label: Name for output csv file containing confusion matrix and accuracies
+#% description: If not given write to standard output
+#%end
+
+#%flag
+#% key: d
+#% description: Description of the accuracies
+#%end
+
+#%flag
+#% key: m
+#% description: Print output as a matrix
+#%end
+
+import os
+import sys
+import csv
+import numpy as np
+import grass.script as grass
+
+
+def print_descriptions():
+    #http://gis.humboldt.edu/OLM/Courses/GSP_216_Online/lesson6-2/metrics.html
+
+    # http://web.pdx.edu/~nauna/resources/9-accuracyassessment.pdf
+    oa = ("Overall accuracy\nNumber of correct plots / total number of plots")
```

Comment:
Since the accuracies are defined in terms of pixels, I suggest to also define OA in terms of pixels; plots is confusing in my opinion

Reply:
Thank you very much for the review :)